superset/models/helpers.py

```diff
@@ -871,6 +872,29 @@ def _process_sql_expression(  # pylint: disable=too-many-arguments
                 raise QueryObjectValidationError(ex.message) from ex
         return expression
 
+    def _should_quote_identifier(self, expr: str) -> bool:
```

Comment:
There's a standard way of doing conditional quoting in `sqlglot`, let me find it.

Reply:
```python
>>> import sqlglot
>>> from sqlglot.dialects.postgres import Postgres
>>> Postgres().quote_identifier(sqlglot.exp.Identifier(this="test"), identify=False)
Identifier(this=test, quoted=False)  # not quoted
>>> Postgres().quote_identifier(sqlglot.exp.Identifier(this="test test"), identify=False)
Identifier(this=test test, quoted=True)  # quoted
>>>
```